src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -236,12 +237,18 @@ public VariantContext finalizeAnnotations(VariantContext vc, VariantContext orig
                     variantAnnotations.putAll(annotationsFromCurrentType);
                 }
                 //clean up raw annotation data after annotations are finalized
-                if (!keepRawCombinedAnnotations) {
-                    variantAnnotations.keySet().removeAll(currentASannotation.getRawKeyNames());
+                for (final String rawKey: currentASannotation.getRawKeyNames()) {
+                    if (!keepRawCombinedAnnotations) {
+                        variantAnnotations.remove(rawKey);
+                    }
                 }
             }
         }
-        variantAnnotations.remove(GATKVCFConstants.AS_QUAL_KEY); //this is manual because the "rawKeys" get added by genotyping
+        //until we refactor the allele-specific annotations to have multiple raw keys, we'll do this one manually
```

Comment:
Comment seems outdated -- you've already refactored the AS annotations to support multiple raw keys, haven't you?

Reply:
Yeah, that's not the real reason.  I've clarified the comments.